src/util.rs

```diff
@@ -46,3 +48,15 @@ impl MsKeyBuilder for script::Builder {
         }
     }
 }
+
+// This belongs in rust-bitcoin, make this upstream later
+pub(crate) fn script_is_v1_tr(s: &Script) -> bool {
+    let mut iter = s.instructions_minimal();
+    if s.len() != 32
```

Comment:
This is bug: the key is 32 bytes indeed, but the script s 34.

```suggestion
    if s.len() != 34
```